Directory.Build.props

```
@@ -16,6 +16,34 @@
     <LangVersion>preview</LangVersion>
     <ImplicitUsings>enable</ImplicitUsings>
     <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
+
+    <IsAnalyzerProject>false</IsAnalyzerProject>
+    <IsAnalyzerProject Condition="!$(IsTestProject) and
+                                  ($(MSBuildProjectName.EndsWith('.Analyzers')) or $(MSBuildProjectName.EndsWith('.Analyzers.CSharp')))"
+                                  >true</IsAnalyzerProject>
+  </PropertyGroup>
+
+  <!--
+    Set assembly version to align with major and minor version, as for the patches and revisions should be manually
+    updated per assembly if it is serviced.
+
+    Note, any components that aren't exposed as references in the targeting pack (like analyzers/generators) those should rev
+    so that they can exist sxs. The compiler relies on different version to change assembly version for caching purposes.
+
+    Because it is possible to build on a lower version and run on a higher version, we're locking the targeting pack and the runtime
+    assembly vesions to 6.0.2.
+    There's still a risk that some customers building on 6.0.2+ and trying to run on 6.0.0 and 6.0.1 will still have issue, but
+    those issues can be mitigated with a workaround described in https://github.com/dotnet/core/issues/7176.
+    -->
+  <PropertyGroup Condition="'$(IsAnalyzerProject)' != 'true'">
+    <AssemblyVersion>$(MajorVersion).$(MinorVersion).2.0</AssemblyVersion>
```

We only want to use the `.2` patch for 6.0... for 7.0 and onwards the ref assemblies should use the `.0` patch.

```suggestion
    <AssemblyVersion>$(MajorVersion).$(MinorVersion).0.0</AssemblyVersion>
    <AssemblyVersion Condition="'$(MajorVersion).$(MinorVersion)' == '6.0'">6.0.2.0</AssemblyVersion>
```

---

This is a 6.0 branch. For the 7.0 branch it is locked at Major.Minor.0.0, see https://github.com/dotnet/winforms/pull/6668.
Is there a situation where you expect 6.0 spill into another branch?

---

In our repos we merge changes forward.  If you don't do that it should be fine, but if that's the case it might as well just be hardcoded to `6.0.2.0` instead of `$(MajorVersion).$(MinorVersion).2.0`.

---

We never merge forward, and always do dual check ins for the release branches.
I'll burn 6.0.2.0 in.